src/service.ts

```diff
@@ -331,6 +332,42 @@ export class Service {
     }
     return uri;
   }
+
+  static downloadProject(project: Project, uri?: string) {
+    const zipFile: JSZip = new JSZip();
+    let zipName: string = "download.zip";
+    if (!isUndefined(uri)) {
+      zipName = `${uri}.zip`;
+    }
+    function defineZip(file: File, folder?: any) {
+      if (file instanceof Directory) {
+        let folder: any;
+        if (!(file instanceof Project)) {
+          folder = zipFile.folder(file.name);
+        }
+        file.mapEachFile((file: File) => defineZip(file, folder));
+      } else {
+        if (!isUndefined(folder)) {
+          folder.file(file.name, file.data);
+        } else {
+          zipFile.file(file.name, file.data);
+        }
+      }
+    }
+    defineZip(project);
```

Comment:
Does `zipFile` have the same interface as the result of `zipFile.folder(..)`? If so, we can pass `zipFile` here as the `folder` parameter, and remove the special case logic in the `defineZip` function.